src/aria2p/api.py

```diff
@@ -70,17 +70,19 @@ def add(self, uri: str) -> List[Download]:  # noqa: WPS231 (not that complex)
 
         if path_exists:
             if path.suffix == ".torrent":
-                new_downloads.append(self.add_torrent(path))
+                new_downloads.append(self.add_torrent(path, options=options))
             elif path.suffix == ".metalink":
-                new_downloads.extend(self.add_metalink(path))
+                new_downloads.extend(self.add_metalink(path, options=options))
             else:
-                for line in read_lines(path):
-                    if line:
-                        new_downloads.extend(self.add(line))
+                for download in self.add_input_file(path):
+                    links, options = download[0], download[1]
+                    for link in links:
+                        new_downloads.extend(self.add(uri=link, options=options))
```

Comment:
You must use `self.add_uris` instead, and only call it once for each group of URIs.